CMakeLists.txt

```diff
@@ -2,11 +2,27 @@
 # Copyright (C) Electronic Arts Inc.  All rights reserved.
 #-------------------------------------------------------------------------------------------
 cmake_minimum_required(VERSION 3.1)
+project(EASTL CXX)
+
+#-------------------------------------------------------------------------------------------
+# Sub-projects
+#-------------------------------------------------------------------------------------------
+add_subdirectory(test/packages/EAAssert)
+add_subdirectory(test/packages/EABase)
+add_subdirectory(test/packages/EAStdC)
+add_subdirectory(test/packages/EAMain)
+add_subdirectory(test/packages/EATest)
+add_subdirectory(test/packages/EAThread)
+
+add_subdirectory(benchmark)
```

Comment:
So what speaks against that options and turning them off by default? No one would see the tests and benchmark pop up in their project unless explicitly enabled.
The point of this PR was to get everything build from the top CMake file and don't have to run CMake from different sub-directories to get the tests and benchmark to build. There needs to be some kind of add_subdirectory call so CMake will process the build files in these directories.

I might miss the point of your concern, so please elaborate if possible.